packages/taquito-tzip16/src/handlers/tezos-storage-handler.ts

```diff
@@ -33,17 +32,17 @@ export class TezosStorageHandler implements Handler {
       parsedTezosStorageUri.contractAddress || contractAbstraction.address,
       'head'
     );
-    const bigMapId = Schema.fromRPCResponse({ script }).FindFirstInTopLevelPair<BigMapId>(
+    const bigMapId = Schema.fromRPCResponse({ script }).FindFirstInTopLevelPair<MichelsonV1ExpressionBase>(
       script.storage,
       typeOfValueToFind
     );
 
-    if (!bigMapId) {
+    if (!bigMapId || !bigMapId.int) {
       throw new BigMapMetadataNotFound();
     }
-
+    // TODO check this
```

Comment:
Consider removing // TODO check this
or adding more detail on what needs to be checked